File: packages/dd-trace/test/llmobs/parity/scenarios/langchain/chat-openai-stream.js
'use strict'

const { finish } = require('../common')
const helpers = require('./helpers')

async function main () {
  const tracer = helpers.loadLangChainTracer()
  const model = helpers.chatOpenAI({ temperature: 0, maxTokens: 16 })
  const stream = await model.stream([['user', 'Stream a short parity response.']])
  for await (const chunk of stream) void chunk
  await finish(tracer)
}

main().catch(error => {
  process.stderr.write(`${error.stack}\n`)
  process.exitCode = 1
})

File: packages/dd-trace/test/llmobs/parity/scenarios/langchain/chat-openai-stream.py
from common import finish

from parity_langchain import chat_anthropic, chat_openai, openai_embeddings, start  # noqa: F401


start()
for _chunk in chat_openai(temperature=0, max_tokens=16).stream([("user", "Stream a short parity response.")]):
    pass
finish()
